h/static/scripts/time.js

```diff
@@ -6,43 +6,79 @@ var day = hour * 24;
 var month = day * 30;
 var year = day * 365;
 
+function lessThanThirtySecondsAgo(date, now) {
+  return ((now - date) < 30000);
+}
+
+function lessThanOneMinuteAgo(date, now) {
+  return ((now - date) < 60000);
+}
+
+function lessThanOneHourAgo(date, now) {
+  return ((now - date) < (60 * 60 * 1000));
+}
+
+function lessThanOneDayAgo(date, now) {
+  return ((now - date) < (24 * 60 * 60 * 1000));
+}
+
+function thisYear(date, now) {
+  return date.getFullYear() === now.getFullYear();
+}
+
+function delta(date, now) {
+  return Math.round((now - date) / 1000);
+}
+
+function nSec(date, now) {
+  return '{} sec'.replace('{}', Math.floor(delta(date, now)));
+}
+
+function nMin(date, now) {
+  return '{} min'.replace('{}', Math.floor(delta(date, now) / minute));
+}
+
+function nHr(date, now) {
+  return '{} hr'.replace('{}', Math.floor(delta(date, now) / hour));
+}
+
+function dayAndMonth(date) {
+  return date.toLocaleDateString(undefined, {day: '2-digit', month: 'short'});
```

Comment:
So we don't forget - it is critical to cache the date formatter using `Intl.DateTimeFormat` (see date-util.js) as it turns out that calling `toLocaleDateString()` on both Chrome and Firefox appears to create a new formatter for each call which is very expensive.